### DISCUSSION TITLE
Discussion should be able to link to issue and versa visa, like issues to issues.

### DISCUSSION THREAD

**Discussion by huan** [community/community#2971] [Discussions]
I'd like to be able to just referring a discussion in an issue and then the issue will be displayed under the discussion, just like the behavior of issues.

**Comment by kookma** [comment]
This is an essential feature! Like Issue and PR any reference, closing /merging and all status shall be reflected on discussion without these features, it is difficult to follow up cases!

**Comment by awvwgk** [comment]
Cross-referencing other discussion, issues or PRs would be a very helpful addition to Discussions. I'm currently missing the possibility to easily find issues or PRs referencing a discussion thread.

**Comment by evi-liu** [comment]
Thank you for the feedback! We are tracking this request on our end and gauging customer impact based on the upvotes and reactions of this post.

**Comment by aredridel** [comment]
This would be great to have between discussions, not just between issues and discussions

**Reply by ShravanSunder** [reply]
yes! 👆🏾

**Reply by mjashanks** [reply]
Please do this 🙏 I am currently doing this manually by adding comments everywhere.

**Reply by aredridel** [reply]
Same!

**Reply by 2jojo** [reply]
yes!

**Reply by michael-hawker** [reply]
Cross-linking is one of the most powerful things on GitHub which helps connect projects and work together, so this really is a big oversight in trying to onboard using discussions and issues together to manage projects.

**Reply by pacifiquem** [reply]
This will help!

**Comment by irh** [comment]
I'd really like to see this happen. I'm seeing the potential for a nice flow of Discussions -> Issues -> PRs, but without autolinking the discussions feel disconnected.

**Reply by wdawson** [reply]
Totally agree with this. It's too easy to lose valuable context from other places. Sometimes there are clear duplicates, but the vast majority of the time there are subtle nuances with each related thing. It would be extremely valuable to be reading the discussion and see other places that referenced it (other Discussions, PRs, Issues). This is preventing my team from converting some issues to discussions because the linked context is so valuable.

**Comment by ShravanSunder** [comment]
a way to link to other discussions from discussions would be great as mentioned above

**Comment by michael-hawker** [comment]
Yeah, I just linked to an issue from a discussion and was very surprised to not see that mention in the issue itself like I would when referencing it from any other issue, pr, or commit... ☹

**Comment by reynolek** [comment]
We are in the process of moving our OSS repo over to discussions, and this topic just came up in our standup as pretty much a blocker.

The loss of context is pretty impactful not only to maintainers but also to community members.

**Reply by huan** [reply]
We decided not to use discussions anymore due to this blocker. sad.

**Comment by ariaieboy** [comment]
any update ?

**Comment by cliffordfajardo** [comment]
+1 for this feature  
 I mentioned this issue of #4301  & it would be nice to this inside of this discussion
 
 


**Mock up:**
<img width="630" alt="CleanShot 2022-01-08 at 03 03 49@2x" src="https://user-images.githubusercontent.com/6743796/148641758-f7e1d888-0912-45f0-a75d-844298ca8812.png">

**Comment by xxchan** [comment]
This feature is critical to make discussion more useful. For example, this discussion https://github.com/risinglightdb/risinglight/discussions/503 talks about a bigger problem which links many issues. Without autolinking, the issues cannot know there's a big discussion there.

**Comment by slog198** [comment]
****

**Comment by hhstore** [comment]
+1

**Comment by dmerand** [comment]
I'm here to support this - my team would greatly benefit from being able to backlink discussion<->discussion, discussion<->issue etc. Without this backlinking it's a big step back from just using issues for discussions. Polls and Q&A are nice but this feature feels incomplete without backlinking.

**Comment by huan** [comment]
Today I raised this thread to the GitHub product team in the Star meeting.

I hope we can hear from the team soon, and looking forward to seeing this new feature being considered seriously in the GitHub product roadmap to make a better community Discussion product on GitHub!

**Reply by entcheva** [reply]
Thank you @huan for bringing this thread up in the GitHub Stars meeting today! We'll discuss with the Discussions team and I'll circle back here soon.

**Reply by danielorbach** [reply]
Any update on this most valuable feature? 

/cc @entcheva @huan @evi-liu

**Reply by marcindulak** [reply]
I'm discovering this thread, and it's already a year after the issue was brought during the meeting. Is there an update @entcheva?

**Reply by marcindulak** [reply]
Github marked this as the answer, and closed the discussion, most likely mistakenly.
The desire from the discussion participants is to reopen this discussion.
Therefore I opened another discussion, asking for this https://github.com/orgs/community/discussions/123987,
and followed up with a general question about closing old, discussions on this forum https://github.com/orgs/community/discussions/128887

Below are more discussions on the topic of auto-linking discussions and issues.
These are newer discussions, linked here only for reference.
https://github.com/orgs/community/discussions/3193
https://github.com/orgs/community/discussions/3211
https://github.com/orgs/community/discussions/4301
https://github.com/orgs/community/discussions/73401 (closed)
https://github.com/orgs/community/discussions/125524 (closed)

**Reply by xihajun** [reply]
it has being 2 years, can we probably get any update for this? @entcheva

**Reply by marcindulak** [reply]
Two years since @entcheva comment will be on November 22 2024.
However, if we wait until December 12 2024, it will be 4 years since @huan opened the discussion.

**Reply by huan** [reply]
😅 I have been following up on a [Google App Script issue](https://issuetracker.google.com/issues/36763437) for almost 10 years. That's why I love startups most of the time: it is straightforward to schedule a meeting with the Founder and CEO, and they will tell you, "No, we will not do that in the near future," with a reason why.

**Reply by marcindulak** [reply]
@LiteBrite82 -  I see the discussion is "Closed", but I believe the feature is not implemented yet.

Is github is still monitoring the interest in this feature, to place it on a roadmap?

**Reply by michael-hawker** [reply]
@LiteBrite82 if the intent is to have a single issue to track, this should be the main parent issue as it has the most comments/scenarios as well as upvotes/thumbs compared to the other issues you marked as answer and linked to... Seems like we're all very confused and very much missing this important feature.

**Comment by slog198** [comment]
Привет Марцин,

Большое спасибо за предоставленные ссылки на новые обсуждения по этой теме.
Я согласен, что возможность автоматического связывания обсуждений между
собой была бы очень полезной функцией. Я добавлю эти ссылки в список
рассмотрения для дальнейшего обсуждения с командой.

С наилучшими пожеланиями,
Александр

**Comment by black7375** [comment]
In some cases, backlinks are required, such as commits following a vote to change a logo or package name.

**Comment by lelamaltez** [comment]
Linking discussions to issues and vice versa can be a valuable feature for project management and collaboration on platforms like GitHub. While GitHub currently supports linking discussions to issues through comments or manual linking, bidirectional linking between issues and discussions directly is not yet a native feature as of my last update in January 2022.

However, you can achieve similar functionality using manual methods or third-party tools:

Manual Linking: You can manually create links between discussions and issues by referencing them in comments or descriptions. For example, in a discussion comment, you can mention the relevant issue using its URL or issue number. Similarly, in an issue description or comment, you can reference a discussion by providing its URL.

Third-Party Tools: There are third-party tools and integrations available that provide more advanced linking capabilities between discussions and issues. These tools often offer features such as bidirectional linking, automatic linking based on keywords, and more seamless integration between discussions and issues.

**Comment by huan** [comment]
Sorry but I cannot understand… what happened here? 🧐

**Reply by irh** [reply]
It's not healthy community management to close discussions without a comment, it's a shame to see that happening here.

I think that this was closed because there's an answer marked that links to similar discussions. #73401 has also been closed, I left a comment on #3211 to try to give it a boost.

**Reply by marcindulak** [reply]
https://github.com/orgs/community/discussions/2971 - the present discussion has most votes, so I would argue it is the one to keep open.

**Reply by LiteBrite82** [reply]
Hi @marcindulak 

Yes, I marked the links to other discussions as the answer and closed the discussion in an attempt to consolidate the topic. This discussion gets overlooked for vote counting as it is 4 years old. Just trying to update it to see if the community is still interested - much has changed in 4 years. 

I can reopen the discussion, but as far as internal data collection goes, I think its better to transition to the newest thread and link the legacy threads to it so the devs can see its still an in demand feature. It will have much higher visibility and will be easier to prioritize with fresh votes.

**Reply by LiteBrite82** [reply]
Looking into other options because I agree with @michael-hawker, its better for the overall discussion for this thread to be the main one. I just want to make sure the data (votes and reactions) isn't being overlooked due to the age of the discussion. 

Doing some research before making another change to the discussion - measure twice, cut once.

**Reply by marcindulak** [reply]
I have some comments, which may help in the decision:

1) If you say that github project managers overlook old problems, which over the years acquired a large number of votes, this is concerning. It's natural that people may keep asking for a feature, but after some time with no response they move on.
This xkcd illustrates it https://xkcd.com/2881/

2) The newer, duplicate discussion https://github.com/orgs/community/discussions/3211, from January 22 2021, is only about a month newer than our https://github.com/orgs/community/discussions/2971, from December 12 2020. I would say the (concerning) approach of focusing on newer issues does not apply here.

3) The lack of the very feature we are asking for, autolinking of issues <-> discussions, and/or discussions <-> discussions, is one of the reasons why it's hard to estimate the real interest in this feature. People may go around and discourage projects from using github discussions due to the autolinking functionality lacking, and recommend sticking to issues. I have personally advised against github discussions a few times, last time in https://github.com/inconshreveable/ngrok/issues/952, but also for internal projects. Without autolinking back to https://github.com/orgs/community/discussions/2971, we cut off one information channel about how many times people abandoned the idea of using discussions.
This can be contrasted to the case of issues, which support autolinking, and the interest in the given functionality is visible also from the number of autolinks and not only upvotes, check for example https://github.com/tcort/markdown-link-check/issues/304 or https://github.com/pallets/markupsafe/issues/284.

4) It is true that some projects use discussions like a kind of waste basket, for beginner questions (see for example https://github.com/miguelgrinberg/microdot/issues/200), or deprioritized issues. To support this use of discussions, it may be possible implement the autolinking functionality such that the behavior is set on a per repository basis, or on a more specific granularity level, like for incoming and outgoing links.

**Reply by marcindulak** [reply]
@LiteBrite82 what is your view on the above points?

**Reply by LiteBrite82** [reply]
I have reopened the discussion

**Reply by LiteBrite82** [reply]
This community is managed by GitHub employees, but it is not a support forum and there is no SLA or guarantee each post will get a response from a community manager or product leader.

If your discussion provided product feedback or a feature request we encourage you to review the "Making Suggestions" [guidelines](https://github.com/community/community#making-suggestions) in the repo readme including: "Once you kick off a discussion, the GitHub product team will evaluate the feedback but will not be able to respond to every submission."

Thank you for contributing and continuing to support other users in the GitHub Community Discussions.

**Reply by marcindulak** [reply]
Thanks, good to hear that older discussions are still evaluated!

**Comment by oriolpiera** [comment]
Hi folks,

A new one in this party.

I already open a discussion (because not related discussion appeared) and inmediatly closed it as seen both previous discussions and the weird thing to close the one-month oldest one.

https://github.com/orgs/community/discussions/125524

I take a seat.

**Comment by laymonage** [comment]
Almost five years later and it's still not here. No updates whatsoever from GitHub since the initial response either.

**Reply by dmerand** [reply]
At this point I consider discussions a dead feature and no longer use them.

**Reply by marythought** [reply]
+1 to desiring this functionality! Right now I'm converting discussion topics to issues using the button, but have to manually link back. Very time consuming.

**Comment by vitorfloriano** [comment]
I guess we will get GTA VI before backlinks to Discussions.

**Comment by jsoref** [comment]
- do you mean https://github.com/community/community/discussions/2971? because, if so, the answer is "you can, as long as you don't use the magic `/orgs/:ORG` thing and instead use the `/:ORG/:REPO` notation.

**Comment by AnilaAnilaN** [comment]


**Reply by marcindulak** [reply]
The requested feature is so pasting a link creates a backlink on the pasted target.

Let me test this, maybe this functionality has been implemented?

First, I'll paste a link to this discussion at another random issue anthropics_claude-code_issues/15892, and I expect a backlink to this issue created in this discussion.

Then, I'll paste here this link of a random issue (this one https://github.com/anthropics/claude-code/issues/15909), and I expect a backling to the discussion to be created at the target issue automatically.

**Reply by marcindulak** [reply]
None of the two actions explained above triggered a backlink, so the requested feature "Discussion should be able to link to issue and versa visa, like issues to issues" does not seem to be implemented.

**Comment by michael-hawker** [comment]
I wanted to inline a poll in an issue, didn't ever really realize that Discussions has polling already... but then without an easy way to link or show that the poll links to an issue, trying to coordinate them isn't that helpful.

This feature would help with that.

**Comment by marcindulak** [comment]
I've just found an earlier request for this feature, at https://github.com/orgs/community/discussions/56 - apparently it was even on a " roadmap to improve the experience" in 2020, but no further details were given.

**Comment by TyceHerrman** [comment]
In addition to the cross-referencing requested in this issue, which now includes PR ↔ Discussion, I'd like to see PR ↔ Discussion relationship also support lifecycle behavior analogous to linked issues so that a PR marked as resolving a Discussion automatically closes that Discussion when merged
